lib/rules/comma-dangle.js

```diff
@@ -13,14 +13,17 @@ const lodash = require("lodash");
 
 /**
  * Checks whether or not a trailing comma is allowed in a given node.
- * `ArrayPattern` which has `RestElement` disallows it.
+ * `ArrayPattern` which has `RestElement` or `ObjectPattern` which has `RestProperty` disallows it.
  *
  * @param {ASTNode} node - A node to check.
  * @param {ASTNode} lastItem - The node of the last element in the given node.
  * @returns {boolean} `true` if a trailing comma is allowed.
  */
 function isTrailingCommaAllowed(node, lastItem) {
-    return node.type !== "ArrayPattern" || lastItem.type !== "RestElement";
+    return !(
+        (node.type === "ArrayPattern" && lastItem.type === "RestElement") ||
+        (node.type === "ObjectPattern" && (lastItem.type === "RestProperty" || lastItem.type === "ExperimentalRestProperty"))
```

Comment:
I'd like to hear others' opinions on this. My preference would be not to do this "just in case" and to add it in when that becomes a reality (or, hopefully, remove the support for "ExperimentalRestProperty" and add in "RestProperty"). If this helps with other parsers that support this syntax (since we do, even though it's still stage 3), then I'm not opposed to it.